TLM/TMPE.API/Notifier.cs

```diff
@@ -0,0 +1,63 @@
+using System;
+
+namespace TrafficManager.API
+{
+    public class OnModifiedEventArgs
+    {
+        /// <summary>
+        /// node/segment (citizen/vehicle in future) that has been modified.
+        /// </summary>
+        public InstanceID InstanceID;
+
+        /// <summary>
+        /// Reserved: the sender that caused the rule to be modified.
+        /// </summary>
+        public object Sender;
+
+        /// <summary>
+        /// Reserved: The rule that has been modified. object type is decided by sender.
+        /// </summary>
+        public object Data;
+    }
+
+    public static class Notifier
```

Comment:
Why this class is static? How you are going to dispose events (remove subscribers)? 
How to handle hot-reload?

Reply:
well TMPE itself does not listen to this. other mods need to listen to this.
I am not sure if hot-reload works on the API assembly. if the code inside API changes, I think we need to restart  CS to test it. I am not sure though.

Reply:
I fixed hot-reload in FPS Booster (testing some other thing before WS update), no issues with hot-swapping API assembly

Reply:
I don't know where to put this though. All the managers have static Instance member.
also `Constants` are not even defined in TMPE API but rather they are in the main TMPE dll.
Where do I put the notifier? should I just make it a singleton to see what happens?